packages/binding-mqtt/src/mqtt-broker-server.ts

```diff
@@ -31,51 +34,38 @@ export default class MqttBrokerServer implements ProtocolServer {
     private port = -1;
     private address: string = undefined;
 
-    private user: string = undefined; // in the case usesername is required to connect the broker
+    private brokerURI: string = undefined;
 
-    private psw: string = undefined; // in the case password is required to connect the broker
+    private readonly things: Map<string, ExposedThing> = new Map<string, ExposedThing>();
 
-    private clientId: string = undefined; // in the case clientId can be used to identify the device
+    private readonly config: MqttBrokerServerConfig;
 
-    private protocolVersion: number = undefined;
+    private broker: mqtt.MqttClient;
 
-    private brokerURI: string = undefined;
+    private hostedServer: Aedes;
+    private hostedBroker: net.Server;
 
-    private readonly things: Map<string, ExposedThing> = new Map<string, ExposedThing>();
+    constructor(config: MqttBrokerServerConfig) {
+        this.config = config;
 
-    private broker: mqtt.MqttClient;
-    private rejectUnauthorized: boolean;
-
-    constructor(
-        uri: string,
-        user?: string,
-        psw?: string,
-        clientId?: string,
-        protocolVersion?: number,
-        rejectUnauthorized?: boolean
-    ) {
-        if (uri !== undefined) {
+        if (config.uri !== undefined) {
             // if there is a MQTT protocol indicator missing, add this
-            if (uri.indexOf("://") === -1) {
-                uri = this.scheme + "://" + uri;
+            if (config.uri.indexOf("://") === -1) {
+                config.uri = this.scheme + "://" + config.uri;
             }
-            this.brokerURI = uri;
+            this.brokerURI = config.uri;
         }
-
-        if (user !== undefined) {
-            this.user = user;
+        if (config.selfHost) {
```

Comment:
in the case of self hosting, the broker uri can be separately provided or the static address in the config can be also used. What is the rationale behind having the two possibilities?

Reply:
Not sure what are you asking. Are you saying that I should peak somehow the `staticAddress` from the CLI config? In this case, it seems that this is handled in the Helpers class and not at the server level (neither HTTP server is using that property).

Reply:
I didn't know that HTTP was not using it properly. It is just weird that if I have am self hosting the broker and have an HTTP binding as well, I need to write twice my IP address.

Reply:
This is relevant only for the generated TDs, the request will arrive properly in any case

Reply:
I think we can resolve this discussion then :)